fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -970,6 +973,25 @@ public long primaryKeySegmentId(@Nonnull String segmentName, boolean create) thr
         return ref.getId();
     }
 
+    public void fileLockSet(byte[] key, byte[] value) {
+        agilityContext.set(key, value);
+        agilityContext.flush();
+    }
+
+    public byte[] fileLockGet(byte[] key) {
+        return context.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_FILE_LOCK, agilityContext.get(key));
+    }
+
+    public void fileLockClear(byte[] key, byte[] value) {
+        agilityContext.accept(aContext -> {
+            final Transaction tr = aContext.ensureActive();
+            // Q: How can we verify that the current fdb value is as expected?
+            tr.addWriteConflictKey(key);
+            tr.clear(key);
```

Comment:
`clear` implies a write conflict.

I am not certain I understand the Q, but if you want to verify that the value of the `key` is `value` at the start of the current transaction, then just `get` it. If it was overwritten and committed and before the current transaction started, then it will not match. It it was overwritten and committed successfully after the current transaction then a subsequent commit of the `clear` will fail. Is that what you need?

Reply:
Yep. That should work.

Reply:
✅